tools/content-generator/src/main/java/org/projectnessie/tools/contentgenerator/cli/DeleteRecursive.java

```diff
@@ -0,0 +1,111 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.tools.contentgenerator.cli;
+
+import static org.projectnessie.model.Content.Type.NAMESPACE;
+
+import com.google.common.base.Preconditions;
+import java.util.Iterator;
+import java.util.List;
+import java.util.stream.Stream;
+import org.projectnessie.client.api.CommitMultipleOperationsBuilder;
+import org.projectnessie.client.api.NessieApiV2;
+import org.projectnessie.error.NessieConflictException;
+import org.projectnessie.error.NessieNotFoundException;
+import org.projectnessie.model.Branch;
+import org.projectnessie.model.ContentKey;
+import org.projectnessie.model.EntriesResponse.Entry;
+import org.projectnessie.model.Operation.Delete;
+import org.projectnessie.model.Reference;
+import picocli.CommandLine;
+import picocli.CommandLine.Command;
+
+/** Deletes a namespace recursively. */
+@Command(
+    name = "delete-recursive",
+    mixinStandardHelpOptions = true,
+    description = "Delete selected namespace and its nested objects (recursively)")
+public class DeleteRecursive extends CommittingCommand {
+
+  @CommandLine.Option(
+      names = {"-r", "--ref", "--branch"},
+      description = "Name of the branch where to make changes (default branch if not set).")
+  private String branchName;
+
+  @CommandLine.Option(
+      names = {"-k", "--key"},
+      paramLabel = "<key element>",
+      description = "Elements (one or more) of a single namespace key to delete.")
+  private List<String> keyElements;
+
+  @Override
+  public void execute() throws NessieNotFoundException, NessieConflictException {
+    try (NessieApiV2 api = createNessieApiInstance()) {
+      if (isVerbose()) {
+        spec.commandLine().getOut().println("Fetching existing entries...");
+      }
+
+      Branch branch;
+      if (branchName == null) {
+        branch = api.getDefaultBranch();
+      } else {
+        Reference reference = api.getReference().refName(branchName).get();
+        Preconditions.checkArgument(
+            reference instanceof Branch, "Reference %s is not a branch.", branchName);
+        branch = (Branch) reference;
+      }
+
+      ContentKey ns = ContentKey.of(keyElements);
+
+      int count = 0;
+      CommitMultipleOperationsBuilder commit = api.commitMultipleOperations();
+      try (Stream<Entry> entries = api.getEntries().reference(branch).prefixKey(ns).stream()) {
+        Iterator<Entry> it = entries.iterator();
+        while (it.hasNext()) {
+          Entry entry = it.next();
+          if (ns.equals(entry.getName())) {
+            Preconditions.checkArgument(
+                NAMESPACE.equals(entry.getType()),
+                "Key %s does not point to a Namespace, it points to %s",
+                entry.getName(),
+                entry.getType());
+          }
+
+          commit.operation(Delete.of(entry.getName()));
```

Comment:
Could it be that we have thousands of child namespaces? Shouldn't we proceed by batches?

Reply:
It is a possibility, but doing batches is kind of complicated because we'd have to sort deleted in the "deep first" manner.

I figured it might be an overkill for this "exploratory" tool. Let me see if it is too hard to implement, though.

Reply:
I'm re-implementing this as a new option for `DeleteContent`... so far looks nicer overall :)

Reply:
I've reworked the impl.... however, the amount of heap need is probably still the same.

At least now the new option is beneficial to other commands that process entries in bulk.

We can certainly limit the amount of in-memory keys at the expense of re-fetching them from the server. Do you think it's worth the effort?